File: leetcode/2441.cpp
#include <iostream>
#include <unordered_set>
#include <vector>

using namespace std;

// Solution Code
class Solution {
 public:
  int findMaxK(vector<int>& nums) {
    int max_k = -1;
    unordered_set<int> waiting_set;
    for (auto num : nums) {
      if (waiting_set.find(num) != waiting_set.end()) {
        max_k = max(max_k, abs(num));
      } else {
        waiting_set.insert(-num);
      }
    }

    return max_k;
  }
};
// ----

int main() {
  Solution sol;

  // Example Input
  vector<int> nums = {3, 2, 4};

  int result = sol.findMaxK(nums);

  cout << result << "\n";
  return 0;
}

File: leetcode/2441.py
"""
https://leetcode.com/problems/largest-positive-integer-that-exists-with-its-negative/
"""
from typing import List, Dict, Optional


class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        waiting_pair: Dict[int, int] = {}
        max_k: int = -1
        for num in nums:
            if num in waiting_pair:
                max_k = max(max_k, abs(num))
            else:
                waiting_pair[-num] = num
        return max_k


def main() -> None:
    sol = Solution()

    # Example Input
    nums: List[int] = [-10, 8, 6, 7, -2, -3]

    result: int = sol.findMaxK(nums)

    # Example Output
    print(f"{result}")


main()
